### ISSUE TITLE
Need an analogous command to m,

### ISSUE THREAD

**Issue by tandrewnichols** [kshenoy/vim-signature#148]
If you use `m,` in a vim script to mark a spot, there's no way to now what mark it used. What about mapping `dm,` to "remove last placed mark." Or is there some way to do this already?

**Comment by kshenoy** [comment]
> there's no way to now what mark it used

You mean apart from the fact that it shows up in the sign column?

`m,` is easy enough to do. It grabs the first mark from a list of all free marks and places it.
The reason why its counterpart `dm,` is not easy to implement is because to do it correctly, you'll have to maintain a history of all the marks that were placed. So that when someone does  `dm,` repeatedly, the correct behavior is observed i.e. marks are removed in order of their recency.

Then there are other things too. Such as if a mark is placed by any other means such as directly placing it then I'll have to ensure that I remove it from my history if it exists etc.

All in all, what I'm saying is this is a lot of work and I'd rather not do it unless there are clear benefits to be had.

**Comment by tandrewnichols** [comment]
I'm specifically talking about when setting marks from other vim scripts where the sign column is of no use. Like if you were creating a mapping that needed to mark the current location, then go somewhere else in the file and modify something, then come back, and you want to clean up the mark you set. You could obviously use something like `mt` or any letter, but there's no way to know in that case if you're overwriting an existing mark. It would be nice to be able to say "use the next available mark" and then later cleanup that mark. One possibility would be, like you said, tracking a history and mapping something to last mark. Another possibility might be calling a function that returned the next available mark (and maybe that exists . . . I have no idea what `m,` does behind the scenes). That would let you do something like:

```viml
let mark = signature#getNextAvailableMark()
normal! exec "m" mark
" . . . and then later
normal! exec "dm" mark
```

**Comment by kshenoy** [comment]
> Like if you were creating a mapping that needed to mark the current location, then go somewhere else in the file and modify something, then come back, and you want to clean up the mark you set. 

Ah, I see what you're saying and how that might be helpful. However, I'm sorry but I don't think I'm going to support deleting the last placed mark because there are a lot of cases that need to be thought through and supported; it's not as simple as it seems on the surface. 

In this case, it might be better to use some other ways to go about it. For the particular case you described, if I understand the intent correctly, you're making edits elsewhere and trying to return to your original location. If that is all you need, then it might be cleaner to use `getpos()` and `setpos()` to save and restore your position.

However, if the intent is not just to return to your original position but also to clean-up after yourself, then this will still not suffice as your jumplist will have changed, your view might have changed, your search history might have changed (depending on how you jump) etc. So if what you need is to make it appear as if nothing has changed then you'll have to put in a little more effort. I have a function to do exactly these sort of things: [Preserve](https://github.com/kshenoy/dotfiles/blob/f639037959fb45cc852c6eba46102723edf2ade7/vim/pack/utils/start/utils/autoload/utils.vim#L88).

**TL;DR**: Using marks to jump back to the original location is not really a good idea. It's akin to using a heavy hammer when more elegant tools exist to do the same thing.
